components/calendar/src/types.rs

```diff
@@ -101,56 +83,21 @@ pub enum YearAmbiguity {
     EraAndCenturyRequired,
 }
 
-/// Information about the era as usable for formatting
-///
-/// This is optimized for storing datetime formatting data.
-#[derive(Copy, Clone, Debug, PartialEq)]
-#[non_exhaustive]
-pub enum FormattingEra {
-    /// An Era Index, for calendars with a small, fixed set of eras. The eras are indexed chronologically.
-    ///
-    /// In this context, chronological ordering of eras is obtained by ordering by their start date (or in the case of
-    /// negative eras, their end date) first, and for eras sharing a date, put the negative one first. For example,
-    /// bce < ce, and mundi < incar for Ethiopian.
-    ///
-    /// The TInyStr16 is a fallback string for the era when a display name is not available. It need not be an era code, it should
-    /// be something sensible (or empty).
-    Index(u8, TinyStr16),
-    /// An era code, for calendars with a large set of era codes (Japanese)
-    ///
-    /// This code may not be the canonical era code, but will typically be a valid era alias
-    Code(Era),
-}
-
-impl FormattingEra {
-    /// Get a fallback era name suitable for display to the user when the real era name is not availabe
-    pub fn fallback_name(self) -> TinyStr16 {
-        match self {
-            Self::Index(_idx, fallback) => fallback,
-            Self::Code(era) => era.0,
-        }
-    }
-}
-
 /// Year information for a year that is specified with an era
 #[derive(Copy, Clone, Debug, PartialEq)]
 #[non_exhaustive]
 pub struct EraYear {
-    /// The era code as used in formatting. This era code is not necessarily unique for the calendar, and
-    /// is whatever ICU4X datetime datagen uses for this era.
-    ///
-    /// It will typically be a valid era alias.
-    ///
-    /// <https://tc39.es/proposal-intl-era-monthcode/#table-eras>
-    pub formatting_era: FormattingEra,
-    /// The era code as expected by Temporal/CLDR. This era code is unique for the calendar
-    /// and follows a particular scheme.
-    ///
-    /// <https://tc39.es/proposal-intl-era-monthcode/#table-eras>
-    pub standard_era: Era,
```

Comment:
Issue: I just remembered that this struct is parallel with the MonthInfo struct in terms of having both a "standard" code and a "formatting" code. Should we think about this holistically?

Reply:
looks like month can be solved with `code: MonthCode, is_leap: bool`?

Reply:
No, the reason they are different is because the Hebrew month Adar has display name "Adar" in normal years and "Adar II" in leap years, but it is the same month code. We weasel around this by making Adar II be a different _formatting_ month code.

Reply:
we have `Date::is_in_leap_year`

Reply:
I'm not sure that works. The data layout is a VZV with 24 entries

```
<M00>  <M01>  <M02>  <M03>  <M04>  <M05>  <M06>  <M07>  <M08>  <M09>  <M10>  <M11>  <M12>
<M00L> <M01L> <M02L> <M03L> <M04L> <M05L> <M06L> <M07L> <M08L> <M09L> <M10L> <M11L> <M12L>
```

We have "Adar" stored in M06 and "Adar II" stored in M06L. However, we have "Shebat" in M05 and "Adar I" in M05L, which _both_ exist in a leap year.

Reply:
The code in `Hebrew::month` works with just the month ordinal and `year_is_leap`, so we can probably just move that into datetime. Might want to add `year_is_leap` to `MonthInfo` so that we don't need a newtype to pass it around in datetime.

Reply:
The code in `Hebrew::month` is very Hebrew specific and doesn't generalize. I think we've re-written that code 3 or 4 times because we kept finding edge cases.